### PR TITLE
PTX-2756 P0 Internal KVDB regression test and validation

### PR DESCRIPTION
**Pull request by nikolaypopov**
Internal KVDB regression tests and P0 validations

Signed-off-by: nikolaypopov <nikolay.popov86@gmail.com>

### CODE REVIEW

File: pkg/util/test/util.go [hunk 1]
```diff
@@ -1110,6 +1110,51 @@ func validateComponents(pxImageList map[string]string, cluster *corev1.StorageCl
...
+		t := func() (interface{}, bool, error) {
+			// Validate KVDB pods
+			listOptions := map[string]string{"kvdb": "true"}
+			podList, err := coreops.Instance().GetPods(cluster.Namespace, listOptions)
+			if err != nil {
+				return nil, true, fmt.Errorf("failed to get KVDB pods, Err: %v", err)
+			}
+
+			desiredKvdbPodCount := 3
+			if len(podList.Items) != 3 {
```
Comment (pure-jliao) [inline pkg/util/test/util.go:1135]:
```suggestion
			if len(podList.Items) != desiredKvdbPodCount {
```

File: pkg/util/test/util.go [hunk 2]
```diff
@@ -1110,6 +1110,51 @@ func validateComponents(pxImageList map[string]string, cluster *corev1.StorageCl
...
+				return nil, true, fmt.Errorf("failed to validate KVDB pod count, expected: %d, actual: %d", desiredKvdbPodCount, len(podList.Items))
+			}
+			logrus.Debugf("Found all %d/%d Internal KVDB pods", len(podList.Items), desiredKvdbPodCount)
+
+			// Validate Portworx KVDB service
+			portworxKvdbServiceName := "portworx-kvdb-service"
+			_, err = coreops.Instance().GetService(portworxKvdbServiceName, cluster.Namespace)
+			if errors.IsNotFound(err) {
```
Comment (pure-jliao) [inline pkg/util/test/util.go:1143]: need to retry for any errors, not NotFound only

File: test/integration_test/basic_test.go [hunk 3]
```diff
@@ -989,6 +997,33 @@ func BasicSecurityRegression(tc *types.TestCase) func(*testing.T) {
...
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*corev1.StorageCluster)
+		require.True(t, ok)
+
+		// Create and validate StorageCluster
+		cluster = ci_utils.DeployAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		// Delete all KVDB pods and validate the get re-created
+		logrus.Info("Delete portworx KVDB pods and validate they get re-deployed")
+		err = coreops.Instance().DeletePodsByLabels(cluster.Namespace, map[string]string{"kvdb": "true"}, 120*time.Second)
```
Comment (pure-jliao) [inline test/integration_test/basic_test.go:1017]: are we able to delete all kvdb pods at once?

**Review by pure-jliao — APPROVED** [review]
LGTM after addressing comments.

File: test/integration_test/basic_test.go (hunk 3 above)
Comment (nikolaypopov) [reply test/integration_test/basic_test.go:1017]: Yes, those are just pods and have nothing to do with an actual KVDB service

File: pkg/util/test/util.go (hunk 2 above)
Comment (nikolaypopov) [reply pkg/util/test/util.go:1143]: Done

File: pkg/util/test/util.go (hunk 1 above)
Comment (nikolaypopov) [reply pkg/util/test/util.go:1135]: Done

**Pull request merged**